wcfsetup/install/files/lib/util/XMLWriter.class.php

```diff
@@ -187,4 +198,37 @@ public function writeAttributes(array $attributes)
             $this->xml->writeAttribute($attribute, $value);
         }
     }
+
+    /**
+     * Set's the encoding for the XML document.
+     *
+     * @param    string    $encoding
+     * @return void
+     * @throws SystemException
+     * @since 6.0
+     */
+    public function setEncoding(string $encoding): void
+    {
+        if ($this->activeDocument) {
+            throw new SystemException('Could not set encoding after document has started.');
```

Comment:
No SystemException in new code, please. `BadMethodCallException` likely is appropriate.